src/main/java/com/redhat/devtools/lsp4ij/features/rename/LSPRenameRefactoringDialog.java

```diff
@@ -192,4 +231,123 @@ protected void dispose() {
+                        String referenceKey = getReferenceKey(reference);
+                        if (referenceKey != null) {
+                            // Only add references we haven't added previously
+                            if (!externalReferencesByKey.containsKey(referenceKey)) {
+                                PsiElement targetElement = reference.resolve();
+                                PsiFile targetFile = targetElement != null ? targetElement.getContainingFile() : null;
+                                TextRange targetTextRange = targetFile != null ? targetElement.getTextRange() : null;
+                                String targetText = reference.getCanonicalText();
+                                // Files match
+                                if ((targetFile != null) && Objects.equals(file, targetFile) &&
+                                        // Text ranges match
+                                        (targetTextRange != null) && Objects.equals(wordTextRange, targetTextRange) &&
+                                        // Text matches according to case-sensitivity
+                                        (caseSensitive ? wordText.equals(targetText) : wordText.equalsIgnoreCase(targetText))) {
+                                    externalReferencesByKey.put(referenceKey, reference);
+                                }
+                            }
+                        }
+                    }
+                    if (progressIndicator != null) {
+                        progressIndicator.checkCanceled();
+                    }
+                    return true;
+                },
+                file.getUseScope(),
+                wordText,
+                UsageSearchContext.ANY,
+                caseSensitive
+        );
+
+        return externalReferencesByKey.values();
+    }
+
+    private static boolean isCaseSensitive(@NotNull LSPRenameParams renameParams,
+                                           @NotNull PsiFile psiFile) {
+        List<LanguageServerItem> languageServers = renameParams.getLanguageServers();
+        if (!ContainerUtil.isEmpty(languageServers)) {
+            // If any supporting language server is case-sensitive, the search must be case-sensitive; it's better to
+            // miss changing things that should have been changed than to change things that should not
+            for (LanguageServerItem languageServer : languageServers) {
+                LSPClientFeatures clientFeatures = languageServer.getClientFeatures();
+                if (clientFeatures != null) {
+                    LSPRenameFeature renameFeature = clientFeatures.getRenameFeature();
+                    if (renameFeature.isRenameSupported(psiFile)) {
+                        LSPCompletionFeature completionFeature = clientFeatures.getCompletionFeature();
+                        if (completionFeature.isCaseSensitive(psiFile)) {

```

Comment:
It is strange to use completion feature for rename. We discussed about this caseSenstive property and perhaps it would be better to move caseSensitive in the clientFeatures (we can do that since wehave not released) and the JSON client configuration content will declare just 

```
{
  "caseSensitive": true
}
```

If you do that, please update JSON Schema docs and description (this caseSensitive is used n completion and rename both).

Reply:
Yes, this is what I meant when that flag was introduced and I said that case-sensitivity is a language/grammar-level setting and not specific to any single feature associated with that language/grammar. I agree that the setting should be up-leveled, particularly before it has ever been included in a release. I'm happy to do that as part of this work.

Reply:
> Yes, this is what I meant when that flag was introduced and I said that case-sensitivity is a language/grammar-level setting and not specific to any single feature associated with that language/grammar.

I am sorry it was difficult for me to understand without usecases. Now I underdtand more with this new usecase.

>  I agree that the setting should be up-leveled, particularly before it has ever been included in a release. I'm happy to do that as part of this work.

Thanks!

Reply:
Oh, not an issue at all. That wasn't meant as an "I told you so." I think we took the right approach of solving the problem we had until we saw another aspect of it. Here we are, and thankfully before the thing that has to be changed has been put into end user hands. No issues at all! I'll get it moved to a feature-independent location in the next day or two.

Reply:
I need to discuss with @fbricon but I think we should create asap a new release to rmbed the fix https://github.com/redhat-developer/lsp4ij/pull/671

If you want we czn merge your PR and I will do the fix in order to embed the PR in the release.